var/spack/repos/builtin/packages/migraphx/package.py

```diff
@@ -38,18 +39,18 @@ def url_for_version(self, version):
     depends_on('protobuf', type='link')
     depends_on('blaze', type='build')
     depends_on('nlohmann-json', type='link')
-    depends_on('py-pybind11', type='build')
     depends_on('msgpack-c', type='link')
     depends_on('half@1.12.0', type='link')
+    depends_on('py-pybind11', type='build')
+    depends_on('py-pybind11@2.6:', type='build', when='@4.1.0')
```

Comment:
```suggestion
    depends_on('py-pybind11@2.6:', type='build', when='@4.1.0:')
```

This way it will also applies to `master` in case you want to build that one too (considering #22943)